lib/nn-syntax/parser.lr

```diff
@@ -747,6 +748,13 @@ constant_Date = {
 
     'start_of' unit:UNIT => new Ast.Value.Date(new Ast.DateEdge('start_of', unit.value));
     'end_of' unit:UNIT => new Ast.Value.Date(new Ast.DateEdge('end_of', unit.value));
+
+    'new' 'DatePiece' '(' year:literal_number ',' month:literal_number ','
+                          day:literal_number ',' hours:literal_number ','
+                          minutes:literal_number ')' => {
```

Comment:
this needs to be a constant_Number, not a literal_number, to pick up NUMBER_ and similar tokens